src/core/handlers/GraphQLHandler.ts

```diff
@@ -165,20 +166,23 @@ export class GraphQLHandler extends RequestHandler<
      * need to parse it since there's no case where we would handle this
      */
     const match = matchRequestUrl(new URL(args.request.url), this.endpoint)
+    const cookies = getAllRequestCookies(args.request)
```

Comment:
We probably don't _need_ to parse cookies here, because we don't have a match, so this shouldn't be useful anywhere, but we do it in the `HttpHandler` so we should probably match that behavior here.

we could return `cookies: {}` in each of the cases of `!match.matches` or `parsedResult === undefined`, and only parse matches. 

we could mirror that check before parsing in `HttpHandler` too I think, but maybe we don't need to bother with that

Reply:
It's okay to keep it here. When we follow through with custom predicate functions (#1804), it would be useful to allow developers to implement GraphQL request matching based on cookies as well. For that, we'd have to always parse cookies and then pass them to the predicate function that determines the match. 

Until we get there, let's leave it as it is right now after your change.